sumologic/resource_sumologic_local_windows_event_log_source.go

```diff
@@ -0,0 +1,151 @@
+package sumologic
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceSumologicLocalWindowsEventLogSource() *schema.Resource {
+	LocalWindowsEventLogSource := resourceSumologicSource()
+	LocalWindowsEventLogSource.Create = resourceSumologicLocalWindowsEventLogSourceCreate
+	LocalWindowsEventLogSource.Read = resourceSumologicLocalWindowsEventLogSourceRead
+	LocalWindowsEventLogSource.Update = resourceSumologicLocalWindowsEventLogSourceUpdate
+	// LocalWindowsEventLogSource.Delete: resourceSumologicSourceDelete
+	LocalWindowsEventLogSource.Importer = &schema.ResourceImporter{
+		State: resourceSumologicSourceImport,
+	}
+
+	// Windows Event Log specific fields
+	LocalWindowsEventLogSource.Schema["log_names"] = &schema.Schema{
+		Type:        schema.TypeList,
+		Required:    true,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Description: "List of Windows log types to collect (e.g., Security, Application, System)",
+	}
+
+	LocalWindowsEventLogSource.Schema["render_messages"] = &schema.Schema{
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     true,
+		Description: "When using legacy format, indicates if full event messages are collected",
+	}
+
+	LocalWindowsEventLogSource.Schema["event_format"] = &schema.Schema{
+		Type:         schema.TypeInt,
+		Optional:     true,
+		Default:      0,
+		ValidateFunc: validation.IntInSlice([]int{0, 1}),
+		Description:  "0 for legacy format (XML), 1 for JSON format",
+	}
+
+	LocalWindowsEventLogSource.Schema["event_message"] = &schema.Schema{
+		Type:         schema.TypeInt,
+		Optional:     true,
+		ValidateFunc: validation.IntInSlice([]int{0, 1, 2}),
+		Description:  "0 for complete message, 1 for message title, 2 for metadata only. Required if event_format is 0",
+	}
+
+	LocalWindowsEventLogSource.Schema["deny_list"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Description: "Comma-separated list of event IDs to deny",
+	}
+
+	// "allowlist":  is not implmemented yet
```

Comment:
can we get the `allow_list` also added?

Reply:
I've done that, tidied up a few loose ends and fixed the two linting bugs